prettyurl.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+# For testing:
+# rm -rf nextcloud-snap
+# git clone --branch prettyurl https://github.com/nextcloud/nextcloud-snap.git
+# cd nextcloud-snap
+
+# Get the correct tar file from the branch itself
+NC_DOMAIN="$(grep 'https://download.nextcloud.com/server' ./snap/snapcraft.yaml | grep -oP 'https://.*')"
+
+# Download the tar file
+curl -fL "$NC_DOMAIN" -o /tmp/ncpackage.tar.bz2
+
+# Extract the archive
+mkdir -p /tmp/nc
+tar -xjf /tmp/ncpackage.tar.bz2 -C /tmp/nc
+
+# Test if the needed file exists
+if ! [ -f /tmp/nc/nextcloud/lib/private/Setup.php ]
+then
+    echo "The Setup couldn't get extracted."
+    exit 1
+fi
+
+# Alway create a new htaccess file
+rm -f ./src/apache/conf/.htaccess
+
+# Get content of the updateHtaccess function and store it in a Variable called updateHtaccess
+updateHtaccess="$(sed -n "/function updateHtaccess/,/function/p" /tmp/nc/nextcloud/lib/private/Setup.php)"
```

Comment:
Because of the mentioned fragility I've come up with this solution so that you will be able to monitor any possible change in the file before you merge it into develop or any stable branch. 
@kyrofa I don't think that upstream will be able to to move all of that into a template file that we can use because they need to modify the template based on your webserver settings and need to append it to the alread existing htaccess file.
So if you should decide that you don't like my aproach, we can probably close this PR now and not in the future since there will be no other way!
But lets see what @kesselb says.

Reply:
Hmm, we might also be able to just run the PHP command that utilizes this instead, have you experimented with that at all?

Reply:
I guess the problem is that the needed function which inserts the lines into the htaccess won't  work without deploying a complete Nextcloud, as far as I understand it and this is honestly nothing I want to look into. To make it work without a Nextcloud instance, we would need to modify the script, too. That's why I thought that modifying the file directly without needing to set up the php stack, etc. would make more sense...

Reply:
And since we can monitor the new htaccess file this way and have confirmed that it currently works, I thought it was feasible and maintainable this way...

Reply:
Ping @kyrofa

Reply:
I don't know if it helps but if I remember correctly Nextcloud Pi has this option as a setting. Maybe you can find something there in the code?

https://ownyourbits.com/nextcloudpi/

https://docs.nextcloudpi.com/en/configuration-reference/#nc-prettyurl

Reply:
No, it doesn't help unfortunately because the snaps .htaccess is read only.